### PR TITLE
Wollah dit is m

### PR DESCRIPTION
**Pull request by Yannickvdv**
Is nog niet compleet natuurlijk. Maar gewoon begin

---
This change is [Reviewable](https://reviewable.io/reviews/maatwork/bar/8)

### CODE REVIEW

**Comment by C-Alexander** [comment]
|`app.ts:21`|
|---|
Require questions twice?

File: models/category.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { Pool } = require('pg');
+
+const pool = new Pool();
+const Logger = require('../models/logger');
+
+
+module.exports.getCategories = function (callback: Function) {
```
Comment (C-Alexander) [inline models/category.ts]: Would work but this isnt the correct naming for Postgresql. The correct table name would be "categories". But this works anyhow

File: routes/categories.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+var express = require('express');
+var router = express.Router();
+var category = require('../models/category');
+
+/* GET categories. */
+router.get('/', function(req, res, next) {
+    category.getCategories(function(error, result){
+        res.render('category', {title: 'Category', msg: error, categories: result.rows});
```
Comment (C-Alexander) [inline routes/categories.ts]: You're rendering "category" but there is no actual page called Category

File: routes/categories.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+var express = require('express');
+var router = express.Router();
+var category = require('../models/category');
+
+/* GET categories. */
+router.get('/', function(req, res, next) {
```
Comment (C-Alexander) [inline routes/categories.ts]: This is not accessible from an HTML app. 
Needs code such as   <code>res.header("Access-Control-Allow-Origin", "*");</code> (can insert your origin there ...)

File: routes/questions.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+var question = require('../models/question');
+
+/* GET questions. */
+router.get('/', function(req, res, next) {
+    question.getQuestion(function(error, result){
+        res.render('questions', {title: 'Questions', msg: error, questions: result.rows});
+    })
+});
+
+module.exports = router;
```
Comment (C-Alexander) [inline routes/questions.ts]: See comment in categories.js

File: models/question.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const { Pool } = require('pg');
+
+const pool = new Pool();
+const Logger = require('../models/logger');
+
+
+module.exports.getQuestions = function (questionID, callback) {
+    const sql = 'SELECT * FROM question, answer WHERE question.questionid = answer.questionid AND question.questionID = $1';
```
Comment (C-Alexander) [inline models/question.ts]: The naming convention for Postgres is to have the tables as plural. So questions, answers. But fontys taught us differently (as we learned mssql) so this is fine for now.

**Comment by C-Alexander** [comment]
In general, why are you sending pages? Don't you just want to send json?

Additionally, we have a folder (/api/) in routes for api functions. Seems more fitting, to me

**Comment by C-Alexander** [comment]
*[app.js, line 15 at r1](https://reviewable.io:443/reviews/maatwork/bar/8#-KwbW7qr02sLRb9pQzE0:-KwbW7qr02sLRb9pQzE1:bb7p34n) ([raw file](https://github.com/maatwork/bar/blob/b4f7eb8f8abfc9c1b42560ede949b2d6cf4bdaef/app.js#L15)):*
> ```JavaScript
> var Logger = require('./models/logger');
> var index = require('./routes/index');
> var register = require('./routes/register');
> ```

test

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maatwork/bar/8)*

**Comment by C-Alexander** [comment]
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app.js, line 19 at r1](https://reviewable.io:443/reviews/maatwork/bar/8#-KwbVnfaGoBsBnjEO0zu:-KwbVnfaGoBsBnjEO0zv:b-pftd7q) ([raw file](https://github.com/maatwork/bar/blob/b4f7eb8f8abfc9c1b42560ede949b2d6cf4bdaef/app.js#L19)):*
> var questions = require('./routes/questions');
> var categories = require('./routes/questions');

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maatwork/bar/8)*

**Comment by C-Alexander** [comment]
pls update

File: models/question.ts (hunk 5 above)
Comment (Yannickvdv) [reply models/question.ts]: Oh never knew that, at fontys they always told me to use singular.

**Review by C-Alexander — CHANGES_REQUESTED** [review]
Please remove app.js

File: routes/api/quizzes.js [hunk 6]
```diff
@@ -1,23 +1,24 @@
 "use strict";
 var express = require('express');
 var router = express.Router();
-var category = require('../../models/category').Category;
+var quiz = require('../../models/qui').Quiz;
```
Comment (C-Alexander) [inline routes/api/quizzes.js]: que? Que si, que no, que dingus dingus dingus

File: routes/api/quizzes.js [hunk 7]
```diff
@@ -1,23 +1,24 @@
 "use strict";
 var express = require('express');
```
Comment (C-Alexander) [inline routes/api/quizzes.js]: delete the JS.

File: routes/api/quizzes.ts [hunk 8]
```diff
@@ -1,26 +1,27 @@
 var express = require('express');
 var router = express.Router();
-var category = require('../../models/category').Category;
+var quiz = require('../../models/qui').Quiz;
```
Comment (C-Alexander) [inline routes/api/quizzes.ts]: Que si, que no

File: routes/api/quizzes.ts [hunk 9]
```diff
@@ -1,26 +1,27 @@
...
+    //Logger.log('error', quiz.getQuestions());
 });
 
-/* POST categories */
+/* POST quizzes */
 router.post('/', function (req, res, next){
     res.header("Access-Control-Allow-Origin", "*");
-    category.create({ title: req.body.title ,description: req.body.description});
+    quiz.create({ title: req.body.title ,description: req.body.description})
+        .catch(error => Logger(error))
```
Comment (C-Alexander) [inline routes/api/quizzes.ts]: pls specify category

File: routes/api/quizzes.ts [hunk 10]
```diff
@@ -1,26 +1,27 @@
 var express = require('express');
```
Comment (C-Alexander) [inline routes/api/quizzes.ts]: use let's, not var's

**Review by C-Alexander — CHANGES_REQUESTED** [review]
Please fix the bug, there's some style problems but those can be survived if needed.

**Pull request merged**